### PR TITLE
Spectrum floating bus addition

### PR DESCRIPTION
**Pull request by TwistedTom**
Added a little bit of the floating bus behaviour, fixes a few games that rely on it.

### CODE REVIEW

**Comment by Pernod70** [comment]
I wasn't aware of the relevance of the spectrum_port_ula_r mirror, which is why I only commented it out rather than remove. With your change though reading odd IO ports never reach expansion devices, none will work, not even Kempston joystick. Rather than return 0xff would it make sense to return m_exp->iorq_r(offset)?
And why the key change for NMI from F12 to BACKSPACE?

**Comment by TwistedTom** [comment]
at the time I submitted, your latest changes were temporarily gone due to bad commit b380514. i'll get up to date and see what's what when I get a minute. most games that use the floating bus trick simply look for the change from ff (beam is in the border area) to anything other than ff, so the ff does need to be returned when appropriate. i guess we need to give the exp device priority somehow when the slot's in use, but just the specific port(s) for the device, not the whole range.
with the old macros, overlapping ranges were possible, i'm not sure how to do it with the new map() functions?
F12 is snapshot by default

**Comment by MoochMcGee** [comment]
Please, for the love of GOD, fix that commit log. Most of these commits are just merging mainline MAME back into your branch which is NOT what you're supposed to do. You're supposed to rebase onto mainline MAME, which results in a MUCH cleaner pull request.

**Comment by MooglyGuy** [comment]
@MoochMcGee You are an external contributor and as such are in no position to speak on behalf of the team. Do not verbally abuse potential contributors or show such an attitude again, or you will find your access to the project curtailed.

Reopening as presumably it was closed due to @TwistedTom being under the mistaken impression that @MoochMcGee speaks, in any way whatsoever, for the MAME team.

**Comment by cuavas** [comment]
@MoochMcGee you need to work on the attitude, seriously.  People can merge upstream onto their PR branches if that's the way they want to work - GitHub makes it easy to preview the net change and squash it into a single non-merge commit.  You also attacked a contributor who was in no way aggressive or abusive.  This isn't the first time you've done this, and you should know better.  You're getting a 6-week ban.

@TwistedTom sorry about this, please ignore @MoochMcGee.  We'd appreciate it if you can work with @Pernod70 to get this PR to a point where we get the improvements without breaking expansion devices.

File: src/mame/drivers/spectrum.cpp [hunk 1]
```diff
@@ -432,9 +432,30 @@ READ8_MEMBER(spectrum_state::spectrum_port_fe_r)
 
 READ8_MEMBER(spectrum_state::spectrum_port_ula_r)
 {
+	/*
+	*  a little bit of the "floating bus" behaviour,
+	*    cpu can peak at the video bus via odd i/o read,
```
Comment (cuavas) [inline src/mame/drivers/spectrum.cpp]: You want the verb "peek" here (not the noun "peak" as in a mountain peak).

**Comment by TwistedTom** [comment]
Opened up-to-date and working replacement for this (#5883), so closing...

**Pull request closed without merging**